checks/raw/security_policy_test.go

```diff
@@ -173,3 +173,191 @@ func TestSecurityPolicy(t *testing.T) {
 		})
 	}
 }
+
+// Test_collectPolicyHits tests the regexes in collectPolicyHits for positive and negative cases.
+func Test_collectPolicyHits(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		input    string
+		expected []checker.SecurityPolicyInformation
+	}{
+		// URL regex
+		{
+			name:  "URL positive",
+			input: "See https://example.com for details.",
+			expected: []checker.SecurityPolicyInformation{
+				{
+					InformationType: checker.SecurityPolicyInformationTypeLink,
+					InformationValue: checker.SecurityPolicyValueType{
+						Match:      "https://example.com",
+						LineNumber: 1,
+						Offset:     4,
+					},
+				},
+			},
+		},
+		{
+			name:     "URL negative",
+			input:    "No links here.",
+			expected: nil,
+		},
+		// Email regex
+		{
+			name:  "Email positive with @",
+			input: "Contact us at security@example.org.",
+			expected: []checker.SecurityPolicyInformation{
+				{
+					InformationType: checker.SecurityPolicyInformationTypeEmail,
+					InformationValue: checker.SecurityPolicyValueType{
+						Match:      "security@example.org",
+						LineNumber: 1,
+						Offset:     14,
+					},
+				},
+			},
+		},
+		{
+			name:  "Email positive with [at] unescaped",
+			input: "Contact us at security[at]example.org.",
+			expected: []checker.SecurityPolicyInformation{
+				{
+					InformationType: checker.SecurityPolicyInformationTypeEmail,
+					InformationValue: checker.SecurityPolicyValueType{
+						Match:      "security[at]example.org",
+						LineNumber: 1,
+						Offset:     14,
+					},
+				},
+			},
+		},
+		{
+			name:  "Email positive with [at] escaped",
+			input: "Contact us at security\\[at\\]example.org.",
```

Comment:
In the README you would be matching this?

`Contact us at security\[at\]example.org.`

Just trying to confirm the level of escaping since you're also escaping the Go here. (p.s. go supports raw string literals.
https://go.dev/ref/spec#String_literals
> Raw string literals are character sequences between back quotes, as in `foo`. Within the quotes, any character may appear except back quote. The value of a raw string literal is the string composed of the uninterpreted (implicitly UTF-8-encoded) characters between the quotes; in particular, backslashes have no special meaning and the string may contain newlines

so this would be equivalent (but not a required change):
```go
input: `Contact us at security\[at\]example.org.`,
```

Reply:
> In the README you would be matching this?
```
Contact us at security\[at\]example.org.
```
Yes, that's my intent. I'm assuming people will be trying to avoid markdown tools misinterpreting that `[at]` as a link. Github-flavored markdown won't, but - I already saw this in a SECURITY.md that I tried scorecard on in the wild.